### PR TITLE
[BULK] - DocuTune - Rebranding of Azure Active Directory to Microsoft Entra terminology (part 2)

### PR DESCRIPTION
**Pull request by alexbuckgit**
Applying Azure AD rebranding changes as announced on July 11, 2023 and outlined in https://aka.ms/AzureADNewName. Changes are part of the Microsoft-wide rebranding effort.

CorrelationId: 2520d58e-25ff-469e-af14-2834d2a1505e
PointOfContact: @CelesteDG

#docutune

### CODE REVIEW

**Comment by alexbuckgit** [comment]
#docutune-draft

File: msal-dotnet-articles/getting-started/choosing-msal-dotnet.md [hunk 1]
```diff
@@ -36,15 +36,15 @@ You're building a desktop or mobile app. Use MSAL.NET directly and start acquiri
 ## Use [Microsoft Identity Web](https://github.com/AzureAD/microsoft-identity-web/)
 
 You're building a confidential client application (Web app, web API, daemon/service app) running on ASP.NET Core, ASP.NET OWIN, or .NET framework/.NET Core. See what Microsoft Identity Web has to offer:
-- Sign in users in web apps in Azure AD application, Azure AD B2C, and CIAM applications
+- Sign in users in web apps in Microsoft Entra application, Azure AD B2C, and CIAM applications
```
Comment (Dickson-Mwendia) [inline msal-dotnet-articles/getting-started/choosing-msal-dotnet.md:39]:
```suggestion
- Sign in users in web apps in Microsoft Entra application, Azure AD B2C, and Microsoft Entra External ID applications
```

File: msal-dotnet-articles/getting-started/choosing-msal-dotnet.md [hunk 2]
```diff
@@ -36,15 +36,15 @@ You're building a desktop or mobile app. Use MSAL.NET directly and start acquiri
...
   - Support Microsoft personal accounts
   - Support guest users
   - Incremental consent and conditional access in web apps
   - Handle SameSite
   - Integrates with "App services authentication"
   - supports PKCE for confidential client applications
   - Brings performant token cache serializers, including distributed
-- Protect web API (with Azure AD, Azure AD B2C or CIAM)
+- Protect web API (with Microsoft Entra ID, Azure AD B2C or CIAM)
```
Comment (Dickson-Mwendia) [inline msal-dotnet-articles/getting-started/choosing-msal-dotnet.md:47]:
```suggestion
- Protect web API (with Microsoft Entra ID, Azure AD B2C, or Microsoft Entra External ID)
```

**Review by Dickson-Mwendia — CHANGES_REQUESTED** [review]
@alexbuckgit I left a minor suggestion here then we can get it merged. Overall, the rebranding updates are looking great. Thanks!

Cc: @CelesteDG

**Comment by localden** [comment]
Looks like `region-discovery-troubleshooting` is missing `description` metadata. Flagging this for future fix, but otherwise LGTM.

**Review by localden — APPROVED** [review]

File: msal-dotnet-articles/advanced/spa-authorization-code.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Today, applications using this architecture will first interactively authenticat
 
 ## Availability
 
-MSAL 4.40+ enables confidential clients to request an additional SPA auth code from the Azure AD token endpoint.
+MSAL 4.40+ enables confidential clients to request an additional SPA auth code from the Microsoft Entra token endpoint.
```
Comment (localden) [inline msal-dotnet-articles/advanced/spa-authorization-code.md:14]: Should this be Microsoft Entra ID?

File: msal-dotnet-articles/getting-started/best-practices.md [hunk 4]
```diff
@@ -22,9 +22,11 @@ ms.custom: devx-track-csharp, aaddev, engagement-fy23
 
 While you can have a look at the contents of an access token (for instance, using https://jwt.ms), for education, or debugging purposes, you should never parse an access token as part of your client code. The access token is only meant for the Web API or the resource it was acquired for. In most cases, web APIs use a middleware layer (for instance [Identity model extension for .NET](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki) in .NET), as this is complex code, about the protection of your web apps and Web APIs, and you don't want to introduce security vulnerabilities by forgetting some important paths.
 
-## Don't acquire tokens from Azure AD too often
+<a name='dont-acquire-tokens-from-azure-ad-too-often'></a>
 
-The standard pattern of acquiring tokens is: (i) acquire a token from the cache silently and (ii) if it doesn't work, acquire a new token from Azure AD. If you skip the first step, your app may be acquiring tokens from Azure AD too often. This provides a bad user experience, because it is slow and error prone as the identity provider might throttle you.
+## Don't acquire tokens from Microsoft Entra too often
```
Comment (localden) [inline msal-dotnet-articles/getting-started/best-practices.md:27]: Microsoft Entra ID?

File: msal-dotnet-articles/how-to/token-cache-serialization.md [hunk 5]
```diff
@@ -378,11 +378,11 @@ var app = ConfidentialClientApplicationBuilder
 
 ## [Desktop apps](#tab/desktop)
 
-In desktop applications, we recommend that you use the cross-platform token cache. MSAL.NET provides the cross-platform token cache in a separate library named [Microsoft.Identity.Client.Extensions.Msal](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet).
+In desktop applications, we recommend that you use the cross-platform token cache. MSAL.NET provides the cross-platform token cache in a separate library named [Microsoft.Identity.Client.Extensions.MSAL](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet).
 
 #### Referencing the NuGet package
 
-Add the [Microsoft.Identity.Client.Extensions.Msal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/) NuGet package to your project.
+Add the [Microsoft.Identity.Client.Extensions.MSAL](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/) NuGet package to your project.
```
Comment (localden) [inline msal-dotnet-articles/how-to/token-cache-serialization.md:385]: This needs to be lowercase. See package name on NuGet.

File: msal-dotnet-articles/how-to/token-cache-serialization.md [hunk 6]
```diff
@@ -539,7 +539,7 @@ static class TokenCacheHelper
  }
 ```
 
-A product-quality, file-based token cache serializer for public client applications (for desktop applications running on Windows, Mac, and Linux) is available from the [Microsoft.Identity.Client.Extensions.Msal](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/tree/master/src/Microsoft.Identity.Client.Extensions.Msal) open-source library. You can include it in your applications from the following NuGet package: [Microsoft.Identity.Client.Extensions.Msal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/).
+A product-quality, file-based token cache serializer for public client applications (for desktop applications running on Windows, Mac, and Linux) is available from the [Microsoft.Identity.Client.Extensions.MSAL](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/tree/master/src/Microsoft.Identity.Client.Extensions.Msal) open-source library. You can include it in your applications from the following NuGet package: [Microsoft.Identity.Client.Extensions.MSAL](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/).
```
Comment (localden) [inline msal-dotnet-articles/how-to/token-cache-serialization.md:542]: Same as previous comment - this needs to be lowercase (same as on NuGet).

**Review by localden — CHANGES_REQUESTED** [review]

**Comment by localden** [comment]
@alexbuckgit @Dickson-Mwendia could you please review my comments? We can't merge without resolving those.

File: msal-dotnet-articles/how-to/token-cache-serialization.md (hunk 5 above)
Comment (Dickson-Mwendia) [reply msal-dotnet-articles/how-to/token-cache-serialization.md:385]: Agreed @localden. The Docutune tool isn't configured to run on code samples, so in most cases it applies documentation rules as per our contributor guide, hence it's suggestion to change to MSAL here, which we shouldn't.

**Review by Dickson-Mwendia — COMMENTED** [review]
@localden, good spot on the incorrect rebranding instances.

File: msal-dotnet-articles/advanced/spa-authorization-code.md (hunk 3 above)
Comment (localden) [reply msal-dotnet-articles/advanced/spa-authorization-code.md:14]:
```suggestion
MSAL 4.40+ enables confidential clients to request an additional SPA auth code from the Microsoft Entra ID token endpoint.
```

File: msal-dotnet-articles/getting-started/best-practices.md (hunk 4 above)
Comment (localden) [reply msal-dotnet-articles/getting-started/best-practices.md:27]:
```suggestion
## Don't acquire tokens from Microsoft Entra ID too often
```

File: msal-dotnet-articles/how-to/token-cache-serialization.md (hunk 5 above)
Comment (localden) [reply msal-dotnet-articles/how-to/token-cache-serialization.md:385]:
```suggestion
Add the [Microsoft.Identity.Client.Extensions.Msal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/) NuGet package to your project.
```

File: msal-dotnet-articles/how-to/token-cache-serialization.md (hunk 6 above)
Comment (localden) [reply msal-dotnet-articles/how-to/token-cache-serialization.md:542]:
```suggestion
A product-quality, file-based token cache serializer for public client applications (for desktop applications running on Windows, Mac, and Linux) is available from the [Microsoft.Identity.Client.Extensions.Msal](https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/tree/master/src/Microsoft.Identity.Client.Extensions.Msal) open-source library. You can include it in your applications from the following NuGet package: [Microsoft.Identity.Client.Extensions.Msal](https://www.nuget.org/packages/Microsoft.Identity.Client.Extensions.Msal/).
```

**Pull request merged**